File: backend/models/item.go
package models

import (
	"time"
)

type Item struct {
	ID         uint     `gorm:"primaryKey" json:"id"`
	Name       string   `gorm:"not null" json:"name"`
	Price      float64  `gorm:"not null" json:"price"`
	CategoryID uint     `json:"category_id"`
	Category   Category `json:"category"`
	CreatedAt  time.Time
	UpdatedAt  time.Time
}

File: frontend/lib/models/item.dart
import 'package:frontend/models/category.dart';

class Item {
  final int id;
  final String name;
  final double price;
  final int categoryID;
  final Category? category;

  Item({
    required this.id,
    required this.name,
    required this.price,
    required this.categoryID,
    required this.category,
  });

  factory Item.fromJson(Map<String, dynamic> json) {
    return Item(
      id: json["id"],
      name: json["name"],
      price: json["price"].toDouble(),
      categoryID: json["category_id"],
      category: json["category"] != null
          ? Category.fromJson(json["category"])
          : null,
    );
  }
}
